### PR TITLE
feat: set `features.use_catalog` to `true` by default

### PR DESCRIPTION
**Pull request by ysndr**
Enable the catalog by default and update generated documentation/manpages.

The pkgdb backend can still be used by configuring `features.use_catalog = false`, however will be deprecated with the catalog release and eventually removed.

### CODE REVIEW

**Comment by mkenigs** [comment]
There are some `unset FLOX_FEATURES_USE_CATALOG` that need to be `export FLOX_FEATURES_USE_CATALOG=false`

File: test_data/config.toml [hunk 1]
```diff
@@ -12,6 +12,13 @@ post_cmd = '''
     mv empty.json "$main_output_dir/empty.json"
 '''
 
+[resolve.failed-resolution]
```
Comment (zmitchell) [inline test_data/config.toml:15]: There's already a convention of using underscores in this file, can we stick to that?

File: test_data/config.toml
Comment (zmitchell) [inline test_data/config.toml:1]: You can also remove the environment variable set in `[vars]`

File: test_data/config.toml (hunk 1 above)
Comment (ysndr) [reply test_data/config.toml:15]: sure, tunnel vision :)

**Comment by zmitchell** [comment]
You've requested review but it's still marked as a draft, is that a mistake?

**Comment by ysndr** [comment]
lets make it ready for review

File: cli/flox/doc/manifest.toml.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 title: MANIFEST.TOML
-section: 5
+section: 1
```
Comment (mkenigs) [inline cli/flox/doc/manifest.toml.md:3]: blocking: this should still be 5, see b92287011

File: cli/flox/doc/manifest.toml.md [hunk 3]
```diff
@@ -329,11 +283,6 @@ bash = """
     alias foo="echo bar"
     set -o vi
 """
-fish = """
-    source $venv_dir/bin/activate.fish
-    alias foo="echo bar"
-    fish_vi_key_bindings
-"""
```
Comment (mkenigs) [inline cli/flox/doc/manifest.toml.md:336]: blocking: bad merge

File: cli/flox/src/config/features.rs [hunk 4]
```diff
@@ -10,7 +10,16 @@ pub struct Features {
...
+    pub use_catalog: UseCatalog,
+}
+
+#[derive(Copy, Clone, Debug, Deserialize, Serialize, derive_more::Deref)]
+pub struct UseCatalog(bool);
+
+impl Default for UseCatalog {
+    fn default() -> Self {
+        UseCatalog(true)
+    }
```
Comment (mkenigs) [inline cli/flox/src/config/features.rs:22]: nonblocking: I think I slightly prefer `#[serde(default="return_true")]`,
```
fn return_true() -> bool {
    true
}
```

Comment (ysndr) [reply cli/flox/src/config/features.rs:22]: Doesn't work unfortunately if we use unwrap_or_default where default is the default impl of bool I.e false

File: cli/flox/doc/manifest.toml.md (hunk 3 above)
Comment (ysndr) [reply cli/flox/doc/manifest.toml.md:336]: i fixed the manpage category and fish, what else got merged badly?,

Comment (mkenigs) [reply cli/flox/doc/manifest.toml.md:336]: That's it 👍

**Review by mkenigs — APPROVED** [review]

**Pull request merged**